### PR TITLE
docs: updated "Getting Started" topic

### PR DESCRIPTION
**Pull request by Doug-AWS**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

I updated the topic based on the feedback from issue #1044; updated code based on V 0.14.1.

### CODE REVIEW

File: docs/src/getting-started.rst [hunk 1]
```diff
@@ -10,43 +10,53 @@
...
+You must install
+`Node.js (>= 8.11.x) <https://nodejs.org/en/download>`_ to use the command-line
+toolkit and language bindings.
+
+You must set the `JAVA_HOME` environment variable to the path to where you have
+installed the JDK on your machine to build an |cdk| app in Java.
+
+.. _setup_recommendations:
+
+Recommendations
```
Comment (eladb) [inline docs/src/getting-started.rst]: This is not really a recommendation. It's required

File: docs/src/getting-started.rst [hunk 2]
```diff
@@ -10,43 +10,53 @@
...
 Prerequisites
 -------------
 
-`Node.js (>= 8.11.x) <https://nodejs.org/en/download>`_ - required for the
-command-line toolkit and language bindings.
+You must install
+`Node.js (>= 8.11.x) <https://nodejs.org/en/download>`_ to use the command-line
+toolkit and language bindings.
+
+You must set the `JAVA_HOME` environment variable to the path to where you have
```
Comment (eladb) [inline docs/src/getting-started.rst]: I would start the sentence with "If you use Java...", so people can "bail out" early if they don't care

File: docs/src/getting-started.rst [hunk 3]
```diff
@@ -59,14 +69,17 @@ Initializing the Project
...
 
-    This guide walks you through the process of creating a |cdk| project
-    step-by-step to explain some of the reasoning and details
-    behind the project structure and tools. It is also possible to use the
-    :code:`cdk init` command to get started quickly from a project
-    template in supported languages.
+    This guide walks you through the process of creating an |cdk| project.
+    You can also use the
+    :code:`cdk init` command to create a skeleton project from a
+    template in any of the supported languages except JavaScript.
```
Comment (eladb) [inline docs/src/getting-started.rst]: That's going to be outdated soon. Maybe we can just write "in supported languages".

File: docs/src/getting-started.rst [hunk 4]
```diff
@@ -93,30 +113,59 @@ Create an empty project structure for the |cdk| app.
...
-                    "watch": "tsc -w"
+                "compilerOptions": {
+                    "target": "es2018",
+                    "module": "commonjs"
                 }
             }
 
     .. group-tab:: Java
 
+        Create an empty source-controlled directory for your project:
```
Comment (eladb) [inline docs/src/getting-started.rst]: Why is this only in java?

File: docs/src/getting-started.rst [hunk 5]
```diff
@@ -93,30 +113,59 @@ Create an empty project structure for the |cdk| app.
...
+            mkdir hello-cdk
+            cd hello-cdk
+            git init
+
+	Create a **.gitignore** file with the following content:
+
+	.. code-block:: sh
+
+	    .classpath.txt
+            target
```
Comment (eladb) [inline docs/src/getting-started.rst]: indentation is off

File: docs/src/getting-started.rst [hunk 6]
```diff
@@ -158,8 +207,7 @@ library includes the basic classes needed to write |cdk| stacks and apps.
 
     .. group-tab:: TypeScript
 
-        Install the **@aws-cdk/cdk** package and the **@types/node** (the latter
-        is needed because we reference **process.argv** in our code):
+        Install the **@aws-cdk/cdk** and the **@types/node** packages:
```
Comment (eladb) [inline docs/src/getting-started.rst]: @types/node is actually not needed anymore

File: docs/src/getting-started.rst [hunk 7]
```diff
@@ -232,25 +280,22 @@ class. Let's create our first, empty **App**:
...
 
-            public class MyApp extends App {
-                public MyApp(final List<String> argv) {
-                    super(argv);
-                }
+            public class MyApp {
+                public static void main(final String argv[]) {
+                    App app = new App();
+
+                    // Create your stacks here as:
```
Comment (eladb) [inline docs/src/getting-started.rst]: why only in java?

File: docs/src/getting-started.rst [hunk 8]
```diff
@@ -181,41 +229,41 @@ library includes the basic classes needed to write |cdk| stacks and apps.
...
-class. Let's create our first, empty **App**:
+|cdk| apps are classes that extend the :py:class:`App <@aws-cdk/cdk.App>`
+class. Create an empty **App**:
 
 .. tabs::
 
     .. group-tab:: JavaScript
 
-        In **index.js**:
+        Create the file **hello-cdk.js** in the **bin** folder:
```
Comment (eladb) [inline docs/src/getting-started.rst]: I think it's clearer if you write it as `bin/hello-cdk.js `

File: docs/src/getting-started.rst (hunk 2 above)
Comment (Doug-AWS) [reply docs/src/getting-started.rst]: will do

File: docs/src/getting-started.rst (hunk 1 above)
Comment (Doug-AWS) [reply docs/src/getting-started.rst]: I'll remove the "Recommendations" heading.

File: docs/src/getting-started.rst (hunk 3 above)
Comment (Doug-AWS) [reply docs/src/getting-started.rst]: done

File: docs/src/getting-started.rst (hunk 5 above)
Comment (Doug-AWS) [reply docs/src/getting-started.rst]: Stoopid editor. I've replaced all tabs.

File: docs/src/getting-started.rst (hunk 6 above)
Comment (Doug-AWS) [reply docs/src/getting-started.rst]: removed

File: docs/src/getting-started.rst (hunk 8 above)
Comment (Doug-AWS) [reply docs/src/getting-started.rst]: sure

File: docs/src/getting-started.rst (hunk 7 above)
Comment (Doug-AWS) [reply docs/src/getting-started.rst]: I'll remove it as we add a stack later.

**Comment by Doug-AWS** [comment]
ping

**Comment by eladb** [comment]
Please merge from master:

```console
$ git fetch origin
$ git merge origin/master
$ git push origin dougsch/test-getting-started
```

**Comment by Doug-AWS** [comment]
Okay, 1 file changed

**Pull request merged**